accelerator/core/serializer/ser_mam.c

```diff
@@ -482,3 +482,82 @@ status_t recv_mam_message_res_serialize(ta_recv_mam_res_t* const res, char** obj
   cJSON_Delete(json_root);
   return ret;
 }
+
+status_t register_mam_channel_req_deserialize(const char* const obj, ta_register_mam_channel_req_t* req) {
+  if (obj == NULL || req == NULL) {
+    ta_log_error("%s\n", ta_error_to_string(SC_SERIALIZER_NULL));
+    return SC_SERIALIZER_NULL;
+  }
+  status_t ret = SC_OK;
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_item = NULL;
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", ta_error_to_string(ret));
+    goto done;
+  }
+
+  json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "seed");
+  if (json_item != NULL && json_item->valuestring != NULL && strlen(json_item->valuestring) == NUM_TRYTES_ADDRESS) {
+    strncpy(req->seed, json_item->valuestring, NUM_TRYTES_ADDRESS);
+    req->seed[NUM_TRYTES_ADDRESS] = 0;
+  } else {
+    ret = SC_SERIALIZER_INVALID_REQ;
+    ta_log_error("%s\n", ta_error_to_string(ret));
```

Comment:
Add human-readable messages.

Reply:
The current error message here is `"Invalid request value in JSON."`.
Should we change it?

Reply:
Once if `seed` can not be deserialized, it is fatal and there must be something wrong. You can enumerate the potential factors.